### ISSUE TITLE
ModelicaInternal: Cached file not closed on line miss in ModelicaStreams_openFileForReading

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1939]
**Reported by beutlich on 2 Mar 2016 12:24 UTC**
In ed6b9a08db42be43341319d2cdf014988a606ba0 for #1104 DS Lund added some kind of caching of the opened text files for reading.

However, the cached file is not closed if there is a line miss. This finally leads to errors like
```
In Modelica.Utilities.Streams.readLine: Not possible to open file "C:\path\to\Examples_readRealParameters.txt" for reading:
Too many open files
```

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1939

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 2 Mar 2016 12:25 UTC**
Error is present since MSL 3.2.1 and also in SVN trunk. I am going to fix it for MSL 3.2.2 only.

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 2 Mar 2016 13:18 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 2 Mar 2016 14:59 UTC**
Resolved by f897237f1294b64231efd6ed77c6c4a5f25e2dc6.

**Comment by modelica-trac-importer** [comment]
**Changelog modified by beutlich on 2 Mar 2016 14:59 UTC**
Fixed repeated opening of cached file in case of line miss in ModelicaStreams_openFileForReading
